### DISCUSSION TITLE
Allow apply a Site constraint to GraphQL schemas

### DISCUSSION THREAD

**Discussion by weotch** [craftcms/cms#8997] [Ideas]
### Description

My use case is an app that uses Sites for publishing workflows.  For example, an app that has dev, uat, and prod sites.  Rather than passing the site in as a query variable, it would be great if a schema could be locked to a Site so public users couldn't easily query non-prod content by switching out the `site` variable.

**Comment by andris-sevcenko** [comment]
Interesting. I like the idea.
